crates/parcel_config/src/partial_parcel_config.rs

```diff
@@ -0,0 +1,693 @@
+                package_name: String::from("b"),
+                resolve_from: Rc::new(PathBuf::from("~")),
+              }])
+              .build()
+              .unwrap();
+
+            assert_eq!(
+              PartialParcelConfig::merge(from, extend),
+              PartialParcelConfigBuilder::default()
+                .$property(vec!(
+                  PluginNode {
+                    package_name: String::from("a"),
+                    resolve_from: Rc::new(PathBuf::from("/")),
+                  },
+                  PluginNode {
+                    package_name: String::from("b"),
+                    resolve_from: Rc::new(PathBuf::from("~")),
+                  },
+                  PluginNode {
+                    package_name: String::from("c"),
+                    resolve_from: Rc::new(PathBuf::from("/")),
+                  }
+                ))
+                .build()
+                .unwrap()
+            );
+          }
+
+          #[test]
+          fn merges_pipelines_with_dot_dot_dot_match_in_grandparent() {
+            let from = PartialParcelConfigBuilder::default()
+              .$property(vec![
+                PluginNode {
+                  package_name: String::from("a"),
+                  resolve_from: Rc::new(PathBuf::from("/")),
+                },
+                PluginNode {
+                  package_name: String::from("..."),
+                  resolve_from: Rc::new(PathBuf::from("/")),
+                },
+                PluginNode {
+                  package_name: String::from("c"),
+                  resolve_from: Rc::new(PathBuf::from("/")),
+                },
+              ])
+              .build()
+              .unwrap();
+
+            let extend_1 = PartialParcelConfig::default();
+            let extend_2 = PartialParcelConfigBuilder::default()
+              .$property(vec![PluginNode {
+                package_name: String::from("b"),
+                resolve_from: Rc::new(PathBuf::from("~")),
+              }])
+              .build()
+              .unwrap();
+
+            assert_eq!(
+              PartialParcelConfig::merge(PartialParcelConfig::merge(from, extend_1), extend_2),
+              PartialParcelConfigBuilder::default()
+                .$property(vec!(
+                  PluginNode {
+                    package_name: String::from("a"),
+                    resolve_from: Rc::new(PathBuf::from("/")),
+                  },
+                  PluginNode {
+                    package_name: String::from("b"),
+                    resolve_from: Rc::new(PathBuf::from("~")),
+                  },
+                  PluginNode {
+                    package_name: String::from("c"),
+                    resolve_from: Rc::new(PathBuf::from("/")),
+                  }
+                ))
+                .build()
+                .unwrap()
+            );
+          }
+        }
+      };
+    }
+
+    test_pipeline_map!(compressors);
```

Comment:
I'd personally write helper functions rather than macro it unless not possible because macros mess-up a lot of IDE features

Reply:
Yeah I get that, but there's far too much repetition / not enough coverage without the use of macros.